auxprogs/filterBam/src/BamToolsAccess.cc

```diff
@@ -138,6 +130,42 @@ uint32_t BamToolsAlignmentRecord::sumDandIOperations() const {
     return sumDandI;
 }
 
+/** returns number of soft clippings through the summation of operations S in the CIGAR string
+ */
+uint32_t BamToolsAlignmentRecord::sumSOperations() const {
+    const std::vector<BamTools::CigarOp> cigar = alignment->CigarData;
+
+    uint32_t sumS = 0;
+
+    // Scanning through all CIGAR operations
+    for (const auto& it : cigar) {
+        // Type means operations, i.e. [MIDNSHPX=]
+        if (it.Type == 'S') {
+            sumS += it.Length;
+        }
+    }
+
+    return sumS;
+}
+
+/** returns number of deletions through the summation of operations D in the CIGAR string
+ */
+uint32_t BamToolsAlignmentRecord::sumDOperations() const {
+    const std::vector<BamTools::CigarOp> cigar = alignment->CigarData;
+
+    uint32_t sumD = 0;
+
+    // Scanning through all CIGAR operations
+    for (const auto& it : cigar) {
+        // Type means operations, i.e. [MIDNSHPX=]
+        if (it.Type == 'D') {
+            sumD += it.Length;
+        }
+    }
+
+    return sumD;
+}
```

Comment:
Couldn't you make this one function with gets a char c = 'D' or 'S'?
Looks like unnecessarily duplicated code.